hw/ip/otbn/dv/otbnsim/sim/state.py

```diff
@@ -188,9 +198,6 @@ def check_jump_dest(self) -> None:
         If not, generates a BadAddrError.
 
         '''
-        if self.pc_next is None:
-            return
-
```

Comment:
I think this changes how we spot code that runs off the top of memory. I *think* it's right, but maybe we can delete some check somewhere else?

Reply:
Really? The `self.get_next_pc() >= self.imem_size` check is still there. 

The `if self.pc_next is None` check limited this function (`check_jump_dest()`) to the cases where we were pc_next was explicitly set (as result of a jump, branch, or loop). After this change, the function runs always, even if pc_next is just pc + 4. Since that's now trivial to do I chose to expand the check as small safety net.

Reply:
> we must have been checking for straight-line code falling off the top of memory in some other way.

Ah, now I understand. I had a look and couldn't find the check, it looks like we didn't check for that before.

Good point about renaming, I did that.

Reply:
Oops! Nice catch :-)